### DISCUSSION TITLE
Deployment help

### DISCUSSION THREAD

**Discussion by kmnahidulislam** [vercel/community#4037] [Help]
### Summary

Hello, I need help for deploying a node app ..
when I render.com to deploy the app , it works perfectly. But when i use vercel.com ,I am facing error , my website deploy successfully but it is not working properly. 
Check the 2 link of my same website from render and vercel :

https://resultjb.vercel.app/

https://resultjb.kmnahidulislam.com 

For render it's work perfectly for back end and for vercel it is not working backend ....
For example..this is a result website... input roll:108001 and registration number: 2013657874 on the website...
It will not work for vercel but will work for render ...
Please help me to deploy perfectly in vercel .. because I love vercel..
This is the file of my result portal:

![Screenshot_2023-09-13-00-30-25-909_com android chrome](https://github.com/vercel/community/assets/144264609/04dcd2a5-9684-49cc-99e8-5f5109a20a9e)

Thanks
-Nahidul

### Steps to Reproduce

Please help me

**Comment by amyegan** [comment]
Sorry you're having trouble with the backend deployment for your site. I get a [405: INVALID_REQUEST_METHOD](https://vercel.com/docs/errors/application-error-codes#405:-invalid_request_method) error when I submit the role and registration numbers.

We need more information to help you figure out what's going wrong. Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)?

**Reply by kmnahidulislam** [reply]
I can't understand what you want to say?

**Reply by kmnahidulislam** [reply]
Now try to enter the link again

**Reply by kmnahidulislam** [reply]
Mam, can you help me?🙏

**Reply by amyegan** [reply]
Sorry I didn't explain well before. I tried the [example](https://github.com/orgs/vercel/discussions/4037#discussioncomment-6987537:~:text=not%20working%20backend%20....-,For%20example..,-this%20is%20a) steps you shared. When I did that, I saw the following lines logged in my browser console:
```bash
cody-widget.js:1     POST https://getcody.ai/widget-backend/9a1dba84-cfcc-4d35-b663-9fa7ff0e4a4e/state 403
cody-widget.js:1     Uncaught (in promise) This domain does not appear to be authorized for this widget.
script.js:40         POST https://resultjb.vercel.app/?roll=108001&reg=2013657874 net::ERR_ABORTED 405
script.js:56         ReferenceError: error is not defined at getResult (script.js:51:28)
```

So, there is a [405](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405) status on the POST request. There's also a [403 Forbidden](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) status on one of your resources which could be related to this problem.

**Comment by kmnahidulislam** [comment]
i have remove getcode.ai chat widget...now fix my problem...look the attached video and read again my post ..

https://github.com/vercel/community/assets/144264609/d657458c-0ae7-4d15-b18b-778a2af8e52c

help me

**Reply by amyegan** [reply]
The 405 error is still the problem. I don't know enough about your app to be able to fix it for you. I would need a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to be able to help you with debugging

**Reply by kmnahidulislam** [reply]
Can you show me the error..? I can't understand what you want to say? And my request is my node app is not working properly in vercel . In render.com /cyclic.sh etc. Platform, my same node app works properly,but in vercel it is not working. I mean it is not working at backend when i submit details it shows no response.in vercel, it shows just the public folder(html page) , not working for node js (index.js,and other) .i seem you understand this case properly and i attached a video my last reply for the error. Please try to reply faster and fix this problem. My website shows me no 405 error from my side. Discuss this problem with your friend please

**Reply by kmnahidulislam** [reply]
Now please help me please

**Reply by amyegan** [reply]
The 405 error returned from `https://resultjb.vercel.app/?roll=108001&reg=2013657874` is the reason that your site does not work. You need to fix that problem.

I do not know anything else about your site code. I cannot offer a specific solution without having more information. If you need help with debugging, please provide a minimal reproducible example following these guidelines: https://vercel.com/support/articles/creating-a-minimal-reproducible-example

**Reply by kmnahidulislam** [reply]
We are again in misunderstanding... because i told you first i use same file render.com and vercel.com ..in this file in frontend html page and in backend nodejs ...but vercel only shows the html page.. that is my problem...you again and again misunderstood 
https://github.com/kmnahidulislam/resultjb.git

This is my full file

**Reply by kmnahidulislam** [reply]
Hello,,,i can't get my solution yet
